### PR TITLE
Replace deprecated :code.lib_dir/2 usages

### PR DESCRIPTION
**Pull request by sabiwara**
Not 100% sure, should we do this replacement following https://github.com/erlang/otp/pull/8091?

### CODE REVIEW

File: lib/elixir/lib/protocol.ex [hunk 1]
```diff
@@ -448,7 +448,7 @@ defmodule Protocol do
   ## Examples
 
       # Get Elixir's ebin directory path and retrieve all protocols
-      iex> path = :code.lib_dir(:elixir, :ebin)
+      iex> path = Path.join(:code.lib_dir(:elixir), "ebin")
```
Comment (josevalim) [inline lib/elixir/lib/protocol.ex:451]: We should use `Application.app_dir` :) here and everywhere else :D

**Review by josevalim — APPROVED** [review]
One suggestions and ship it!

File: lib/elixir/lib/protocol.ex (hunk 1 above)
Comment (sabiwara) [reply lib/elixir/lib/protocol.ex]: Looks much nicer indeed! [fc6d026](https://github.com/elixir-lang/elixir/pull/13395/commits/fc6d02639a114cf15fe6b3f29d1d627b24c8907d)

**Review by josevalim — APPROVED** [review]

**Comment by sabiwara** [comment]
First time I see this CI failure https://github.com/elixir-lang/elixir/actions/runs/8164336936/job/22319403763?pr=13395 🤔
Is it related to this [change](https://github.com/elixir-lang/elixir/pull/13395/files#diff-f4f6d25a3fecb5fb56e9c5f384b0f417310d793f65266c6b64dff3c400bef36bR147) or just flake?

**Comment by josevalim** [comment]
It is a flake. There is something in the helpers file that is leading to non-determinism but I have been unable to debug it. We need to change the script to print more debug when it happens.

**Review by whatyouhide — APPROVED** [review]
💯

**Pull request merged**